src/EmojiList.js

```diff
@@ -1,19 +1,28 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import ItemComponent from './EmojiComponent';
+import SettingsModeButtons from './SettingsModeButtons';
 
 const apply = connect(
-    state => ({ smiles: state.items.smiles })
+    state => ({
+        smiles: state.items.smiles,
+        mode: state.pages.mode
+    })
 )
 
 class EmojiList extends Component{
 
     render() {
-        console.log("EmojiList ", this.props);
+        let { smiles, mode } = this.props;
         return(
             <div>
-                {this.props.smiles.map(item => {
-                    return <ItemComponent text={item.text} id={item.id} />
+                {smiles.map(item => {
+                    return (
```

Comment:
If you just return element it is better to avoid `return` word, especially if you are allowed to do it. Just use it like this:
```javascript
message => (
  <button>{message}</button>
)
```
Parentheses are a good practice as well because it allow you to indent HTML code as it is in document, not in JS code.

Reply:
fixed